diagnose_ceph/troubleshoot_ceph.py

```diff
@@ -0,0 +1,285 @@
+import json
+import optparse
+import os
+import paramiko
+import re
+import subprocess
+import sys
+
+from helpers.exceptions import (SSHCredsNotFoundError, ConnectionFailedError,
+                                TimeoutError, InitSystemNotSupportedError,
+                                JujuInstallationNotFoundError)
+from helpers.decorators import timeout
+from helpers.helpers import MyStr
+
+
+class JujuCephMachine(object):
+    def __init__(self, name, id, public_addr, hostname, has_osd=False,
+                 has_mon=False, internal_ip=None):
+        self.name = name
+        self.id = id
+        self.public_addr = public_addr
+        self.hostname = hostname
+        self.has_osd = has_osd
+        self.has_mon = has_mon
+        self.internal_ip = internal_ip
+
+
+class TroubleshootCeph(object):
+    '''
+        TroubleshootCeph Class to be called to diagnose a ceph cluster
+    '''
+    GOOD_HEALTH = ['HEALTH_OK']
+    BAD_HEALTH = ['HEALTH_WARN']
+    CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
+    init_script = CURRENT_DIR + '/scripts/check_init.sh'
+    arch_script = CURRENT_DIR + '/scripts/find_processor_architecture.sh'
+
+    def __init__(self):
```

Comment:
This init function is getting really large.  Would it be possible to break it up a little?